drivers/storage/portworx/portworx.go

```diff
@@ -745,31 +761,20 @@ func (p *portworx) validateFACDTopology(cluster *corev1.StorageCluster) error {
 		}
 	}
 
-	if facdTopologyEnv == "true" {
-		// FACD topology is enabled. See if we already have the annotation marking this as valid, or if not, check if this is a fresh install
-
-		// TODO: once we support upgrading to FACD topology, it will also be valid after whatever version supports that
-		isValid := false
-		if pxutil.IsFreshInstall(cluster) {
-			if cluster.Annotations == nil {
-				cluster.Annotations = make(map[string]string)
-			}
-			cluster.Annotations[pxutil.AnnotationFACDTopology] = "true"
-			isValid = true
-		}
-		if !isValid {
-			if a, ok := cluster.Annotations[pxutil.AnnotationFACDTopology]; ok && a != "" {
-				isValid = true
-			}
-		}
+	if facdTopologyEnv != "true" {
+		return nil
+	}
 
-		if !isValid {
-			// Fail the request!
-			return fmt.Errorf("FACD topology cannot be enabled on existing clusters")
+	// FACD topology is enabled. See if we already have the annotation marking this as valid.
+	// TODO: once we support upgrading to FACD topology, it will also be valid after whatever version supports that
+	if cluster.Annotations != nil {
```

Comment:
nit: We don't need the nil check. Golang allows you to query a nil map.
The following line won't fail even if `cluster.Annotations` are nil and would return the correct output.
```
if v, ok := cluster.Annotations[pxutil.AnnotationFACDTopology]; ok && v == "true" { ... }
```

Reply:
Oh good point, I was nervous cuz I'd hit an "assignment to entry in nil map" panic before, but that's assignment, not reading.